python/cudf/cudf/core/column/string.py

```diff
@@ -3989,19 +3989,23 @@ def _mimic_inplace(self, other_col, inplace=False):
         return out
 
     def view(self, dtype):
+        if self.null_count > 0:
+            raise ValueError("Can not produce a view of a column with nulls")
         dtype = np.dtype(dtype)
-        if (
-            self.base_children[0][self.offset]
-            * self.base_children[1].dtype.itemsize
-        ) % dtype.itemsize:
-            raise TypeError(
-                "Slice data size must divide evenly into desired dtype size"
-            )
-        else:
-            offset = self.base_children[0][self.offset] * dtype.itemsize
-            return column.build_column(
-                self.base_children[1].data, dtype=dtype, offset=offset
-            )
+        str_byte_offset = self.base_children[0][self.offset]
+        char_dtype_size = self.base_children[1].dtype.itemsize
+        n_bytes_to_view = (
+            self.base_children[1].size - str_byte_offset
+        ) * char_dtype_size
```

Comment:
I think we need `self.base_children[0][self.offset + self.size]` to determine the end bytes of the the characters as opposed to `self.base_children[1].size`. I'm pretty sure the latter is just the base size.

Reply:
Should also add a test of viewing a string column with both a start and end slice.

Reply:
Oh whoops you're right I think this might not include all of the last element if it's not just a single byte. Looking into it.

Reply:
Updated. I think I have the string test in there, should be one for the start, one for the end and one for the middle.